src/veins/modules/mobility/traci/subscriptionManagement/TraCISubscriptionManager.h

```diff
@@ -0,0 +1,218 @@
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+//
+
+#ifndef SRC_VEINS_MODULES_MOBILITY_TRACI_SUBSCRIPTIONMANAGEMENT_TRACISUBSCRIPTIONMANAGER_H_
+#define SRC_VEINS_MODULES_MOBILITY_TRACI_SUBSCRIPTIONMANAGEMENT_TRACISUBSCRIPTIONMANAGER_H_
+
+#include "veins/modules/mobility/traci/subscriptionManagement/PersonSubscriptionManager.h"
+#include "veins/modules/mobility/traci/subscriptionManagement/SimulationSubscriptionManager.h"
+#include "veins/modules/mobility/traci/subscriptionManagement/TraCIPerson.h"
+#include "veins/modules/mobility/traci/subscriptionManagement/TraCITrafficLight.h"
+#include "veins/modules/mobility/traci/subscriptionManagement/TraCIVehicle.h"
+#include "veins/modules/mobility/traci/subscriptionManagement/TrafficLightSubscriptionManager.h"
+#include "veins/modules/mobility/traci/subscriptionManagement/VehicleSubscriptionManager.h"
+#include "veins/modules/mobility/traci/TraCIBuffer.h"
+#include "veins/modules/mobility/traci/TraCIConnection.h"
+#include "veins/modules/mobility/traci/TraCICommandInterface.h"
+
+namespace veins {
+namespace TraCISubscriptionManagement {
+
+/**
+ * @class TraCISubscriptionManager
+ *
+ * This manages all the subscriptions that were made and provides an API
+ * to the underlying subscription managers and request their results.
+ *
+ * @author Nico Dassler <dassler@hm.edu>
+ */
+class TraCISubscriptionManager {
+
+public:
+
+    /**
+     * Default constructor.
+     *
+     * @param explicitUpdateIfSubscriptionUnavailable In case there is no subscription response
+     * for a certain type of participant it will be replaced with an explicit request.
+     */
+    TraCISubscriptionManager(bool explicitUpdateIfSubscriptionUnavailable = true);
+
+    /**
+     * Default destructor.
+     */
+    virtual ~TraCISubscriptionManager() = default;
+
+    /**
+     * Process a subscription result.
+     *
+     * Do not modify the buffer before inserting it into this method.
+     *
+     * This updates the internal state of this object and changes
+     * responses to other methods of this object.
+     *
+     * @param buffer TraCIBuffer containing subscriptions.
+     */
+    void processSubscriptionResult(TraCIBuffer& buffer);
+
+    /**
+     * Returns a list of persons that are new or updated. It is up
+     * to the caller to decide which person is updated and which is
+     * new.
+     *
+     * Multiple calls to this method will not yield the same result. It
+     * will always give you the new or updated persons since the last
+     * time you called this method.
+     *
+     * @return list<TraCIPerson> a list of persons.
+     */
+    std::list<TraCIPerson> getUpdatedPersons();
```

Comment:
This and all other getters should just return a copy (idally a vector) and be const.

Reply:
Also you might want to consider using `std::vector`, there are few use-cases where a `std::list` should be preferred.

Reply:
done.

Reply:
Alright.